corehq/apps/app_manager/templates/app_manager/partials/modules/module_view_settings.html

```diff
@@ -267,6 +267,22 @@ <h4 class="panel-title panel-title-nolink">{% trans "Advanced" %}</h4>
               </div>
             {% endif %}
 
+            {% if request|toggle_enabled:'CASE_LIST_OPTIMIZATIONS' and not module.is_surveys %}
+              <div class="form-group">
+                <label class="control-label {% css_label_class %}">
+                  {% trans "Show Case List Optimizations" %}
+                  <span class="hq-help-template"
+                        data-title="{% trans "Case List Optimizations" %}"
+                        data-content="{% blocktrans %}Set caching and lazy loading for case list columns for better performance. Once saved, reload this page to reveal/hide relevant column on case list. {% endblocktrans %}"
```

Comment:
Unfortunately, this needs a reload.

The way javascripts are aligned for "settings" tab and the details tabs ("case list" and "case detail"), I could not see a way to have this happen without a reload.